### DISCUSSION TITLE
Is the Interface Filter of an M2M targeted to the right collection?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#15970] [Archive]
On `directus_users` collection I have a field `roles` (not role) targeted to `directus_roles` ( to gather multiple roles for the user ).
Adding a `directus_roles.role` to `directus_users.roles` has to be, in my case, on a subset of `directus_roles`.

So I build an Interface filter.
On the dropdown, I was expecting to see the `directus_roles`'fields to filter on but the fields shown are those of  `directus_users`. 
For example:
```
{
	"_and": [
		{
			"roles": {
				"directus_roles_id": {
					"name": {
						"_contains": "SMTHG"
					}
				}
			}
		}
	]
}

```
`roles` belongs to `directus_users`

Apparently, Directus also expects the filter to be on the target collection `directus_roles`. 
I say that, because, after I set the above filter based on `directus_users`, I get the error
`"directus_roles.roles" is not a relational field`
which shows that Directus has correctly attached the filter to `directus_roles` 
but with a `directus_users` field.

Is it possible to filter on the **target** items shown for choice when adding an M2M item?

**Comment by connorwinston** [comment]
I believe you are correct from my testing, the interface filter should show the fields of the target collection of the m2m relation but it is showing the current collections fields.

**Reply by Pachat** [reply]
Thanks. Should I open an issue?

**Reply by connorwinston** [reply]
I created one, you can track it here #15994 !
